### DISCUSSION TITLE
Render markdown based on query params

### DISCUSSION THREAD

**Discussion by fayvori** [mkdocs/mkdocs#3480] [Q&A] [answered]
Hi!

I have the following case: render markdown page based on query params, for example `localhost:8000/?name=Aden`, and in markdown like `Hello, {name}`. I'm using `on_page_markdown` hook and `str.format()`, but cannot find any info about extracting query params from current url. `page.url` returns only relative path of the page. Is there any way to do it ?

Thanks!

**Comment by oprypin** [comment, marked as answer]
MkDocs just produces a bunch of HTML files that don't *do* anything. Then how could an HTML file call back to its origin to use Python code that renders Markdown? When talking about MkDocs, you should just forget that the site happens to be served from a server: if you can't do something with a plain HTML file locally then you also can't do it on a server that just displays HTML files from MkDocs.

**That being said** of course you can do this stuff separately from MkDocs - either with JavaScript or by augmenting the HTML files in whatever way you want by writing separate code that executes on your server of choice (instead of just the typical static serving functionality)
